### PR TITLE
perf: reduce branches

### PR DESCRIPTION
**Pull request by anonrig**
As Bob Marley said; every little thing is gonna be alright.

## Before

```
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Benchmark           Time             CPU   Iterations        GHz cycle/byte cycles/mime instructions/byte instructions/cycle instructions/mime instructions/ns     mime/s    ns/mime      speed  time/byte  time/mime
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
BasicBench      72347 ns        72346 ns         9677    3.27756    23.7897     268.168           128.236            5.39037          1.44553k         17.6672 12.1775M/s    81.8195  137.27M/s  7.28492ns  82.1186ns

```

## After

```
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Benchmark           Time             CPU   Iterations        GHz cycle/byte cycles/mime instructions/byte instructions/cycle instructions/mime instructions/ns     mime/s    ns/mime      speed  time/byte  time/mime
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
BasicBench      68337 ns        68337 ns        10265    3.27956    22.4421     252.976           119.032            5.30399          1.34178k         17.3947  12.892M/s    77.1373 145.324M/s   6.8812ns  77.5677ns

```

### CODE REVIEW

**Review by KhafraDev — APPROVED** [review]

**Comment by anonrig** [comment]
@KhafraDev @lemire The last commit I've pushed for adding collecting code points, was missing from our code, which was adding unnecessary overhead (and re-iteration)

File: src/parser.cpp [hunk 1]
```diff
@@ -1,3 +1,4 @@
+#include <iostream>
```
Comment (lemire) [inline src/parser.cpp]: Do you need iostream?

File: include/ada/mimesniff/util-inl.h [hunk 2]
```diff
@@ -95,47 +95,46 @@ constexpr inline bool contains_only_http_quoted_string_tokens(
   return true;
 }
 
-inline std::string collect_http_quoted_string(std::string_view input,
-                                              size_t& position) {
+inline std::string collect_http_quoted_string(std::string_view& input) {
   std::string value{};
 
   // TODO: Assert: the code point at position within input is U+0022 (").
```
Comment (lemire) [inline include/ada/mimesniff/util-inl.h]: I wouldn't have this TODO. Just document in the function that the caller is responsible for having checked that the string begins with `"`.

File: src/parser.cpp (hunk 1 above)
Comment (anonrig) [reply src/parser.cpp]: No, you're right. Removed it, and updated the benchmarks :-)

File: include/ada/mimesniff/util-inl.h [hunk 3]
```diff
@@ -95,47 +95,46 @@ constexpr inline bool contains_only_http_quoted_string_tokens(
...
 
       // Append the code point at position within input to value.
-      value += input[position];
+      value += input[0];
 
       // Advance position by 1.
-      position++;
+      input.remove_prefix(1);
     } else {
       // TODO: Assert: quoteOrBackslash is U+0022 (").
```
Comment (lemire) [inline include/ada/mimesniff/util-inl.h]: Well. I don't think an assert (in the conventional sense) is warranted here. 

Note that you could probably simplify the code like so...

```C++
  if(input[0]  == '"') {
    break;
  }
 // we know that (input[0]  == '\\')
```

Comment (anonrig) [reply include/ada/mimesniff/util-inl.h]: You're right.

**Pull request merged**